LayoutTests/media/media-source/media-source-first-sample-pts-non-zero.html

```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<title>media-source-first-sample-pts-non-zero</title>
+<style type="text/css">
+    video {
+        width: 150px;
+        height: 150px;
+    }
+    .zeroSize {
+        width: 0px;
+        height: 0px;
+    }
+</style>
+</head>
+<body>
+<p>This test verifies that the "canplay" event fires on a video element with a MediaSource that loads a video track whose first sample has a presentation timestamp greater than 0.</p>
+<script type="text/javascript">
+    if (window.testRunner)
+        testRunner.waitUntilDone();
+
+    var source = new MediaSource();
+    source.addEventListener("sourceopen", async event => {
+        var buffer = source.addSourceBuffer("video/mp4");
+        var response = await fetch("resources/first-sample-pts-non-zero/first-sample-pts-non-zero.mp4");
+        buffer.appendBuffer(await response.arrayBuffer());
+    });
+    var video = document.createElement("video");
+    video.classList.add("zeroSize");
+    document.body.appendChild(video);
+    video.addEventListener("canplay", event => {
```

Comment:
a bit more modern JS would be to make this async and do:
```
await waitFor(video, 'canplay');
testRunner.notifyDone();
```
(make this into a onload=test())